src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -529,8 +529,8 @@ private LocalSymbol MakePatternVariable(TypeSyntax type, VariableDesignationSynt
             }
 
             NamedTypeSymbol container = _scopeBinder.ContainingType;
-            if ((object)container != null && container.IsScriptClass &&
-                (object)_scopeBinder.LookupDeclaredField(designation) != null)
+            if (container is { IsScriptClass: true } &&
+                (object)_scopeBinder.LookupDeclaredField(designation) is { })
```

Comment:
similar complaint.  if the expression is getting completely pattern-ized, tehn i'm ok with `{}`.  but this expression was already fine and the trivial change from the clear `!= null` to `is { }` i don't think benefits the code.